### PR TITLE
Marine Body Spray AKA: Reworks Marine Orders

### PR DESCRIPTION
**Pull request by CometBlaze**
# About the pull request

SLs are now able to pick between three kinds of body spray laced with experimental stimulants designed to improve the marines' combat performance.

On a more serious note this makes it so orders now act as  a buff aura rather than a short power boost. In order to compensate for their now constant nature, the strength of the mobility and protection auras have been halved, focus has been left as is because it was already weak enough and honestly I would still like to make it stronger, but maybe in another PR.

For the majority of roles the changes end here, however, squads will see another aspect changed in the form of their FTLs. When an SL (or aSL) picks an order, all FTLs currently assigned to that squad will also emit that same aura. This aura has the same exact behaviour as the regular SL's orders with the only difference being that FTLs cannot pick their own orders.

Assuming the maintainer team doesn't shank me for how I coded this there will be a follow up to this PR introducing ~fear RP~ morale de-buffs to marines not within range of a "buff piece" (SL, FTL, CO, etc...)

# Explain why it's good for the game

Provides marine with a mechanical incentive to stick with their SLs which should hopefully lead to better squad cohesion.
Increased squad cohesion should allow for more elaborate tactics to be employed by CIC which should lead to more interesting rounds.
Benefits to squad cohesion also apply to Fireteams, which should allow for even more flexibility in squad tactics.
# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

https://github.com/cmss13-devs/cmss13/assets/59415394/58822eb9-2a6a-4533-9edd-5e71dac03f3b

</details>

# Changelog
:cl: CometBlaze
add: Orders are now buff auras that apply to anyone within 7 tiles of the buff giver and no longer have limited duration. 
balance: Mobility and Protection orders' strength have been halved, Focus remains unchanged.
add: FTLs now act as "order relays" for their SLs, meaning they will also have an aura like the SL, this aura has the same properties as the SL's and remains active for as long as theirs is.
code: Order action buttons have been brought into the modern age and now use a transparent action_icon_state
/:cl:

### CODE REVIEW

**Comment by Ben10083** [comment]
How does this affect defence placements like the warbanners?

**Comment by ihatethisengine** [comment]
I think spray should affect only their own squad, to discourage murderballing a bit (otherwise imagine Alpha Charlie Delta SLs sticking together to provide every kind of pheros). And FTL should relay orders only to their respective TL. I think this way it would be perfect.

**Comment by Weuyn** [comment]
really prefer the current system to passive buffs. Passives aren't as fun

**Review by sleepynecrons — APPROVED** [review]
sprites

**Review by nauticall — APPROVED** [review]
concur, sprites good

File: code/modules/mob/living/carbon/human/human_defines.dm [hunk 1]
```diff
@@ -93,11 +93,13 @@
...
+	//var/command_aura_available = TRUE // Whether or not you can issue an order
+	var/current_aura = null //The aura we're currently emitting
+	var/aura_strength = 0 //Strenght of the aura we're emitting
 
-	var/mobility_aura_count = 0 //Used to track how many auras are affecting the human
-	var/protection_aura_count = 0
-	var/marksman_aura_count = 0
+	var/mobility_aura_new = 0 //Used to track how many auras are affecting the human
+	var/protection_aura_new = 0
+	var/marksman_aura_new = 0
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/human_defines.dm:102]: code doc these 5 (or more, if you want)

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 2]
```diff
@@ -408,7 +408,7 @@ This function restores all limbs.
 	permanent_kill = FALSE, mob/firer = null, force = FALSE
 )
 	if(protection_aura && damage > 0)
-		damage = round(damage * ((ORDER_HOLD_CALC_LEVEL - protection_aura) / ORDER_HOLD_CALC_LEVEL))
+		damage = round(damage * ((ORDER_HOLD_CALC_LEVEL - protection_aura * 0.5) / ORDER_HOLD_CALC_LEVEL))
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/human_damage.dm:411]: are you certain this is doing what you want with PEMDAS?

File: code/modules/mob/living/carbon/human/powers/issue_order.dm [hunk 3]
```diff
@@ -7,55 +7,58 @@
...
 		order_level = SKILL_LEAD_TRAINED
 
 	if(!order)
-		order = tgui_input_list(src, "Choose an order", "Order to send", list(COMMAND_ORDER_MOVE, COMMAND_ORDER_HOLD, COMMAND_ORDER_FOCUS, "help", "cancel"))
-		if(order == "help")
-			to_chat(src, SPAN_NOTICE("<br>Orders give a buff to nearby soldiers for a short period of time, followed by a cooldown, as follows:<br><B>Move</B> - Increased mobility and chance to dodge projectiles.<br><B>Hold</B> - Increased resistance to pain and combat wounds.<br><B>Focus</B> - Increased gun accuracy and effective range.<br>"))
+		if(current_aura)
+			deactivate_order_buff(current_aura)
+			current_aura = null
+			visible_message(SPAN_WARNING("\The [src] stops issueing orders."), \
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/powers/issue_order.dm:20]: issuing

File: code/modules/mob/living/carbon/human/powers/issue_order.dm [hunk 4]
```diff
@@ -7,55 +7,58 @@
...
+			order = tgui_input_list(src, "Choose an order", "Order to send", list(COMMAND_ORDER_MOVE, COMMAND_ORDER_HOLD, COMMAND_ORDER_FOCUS, "help", "cancel"))
+			if(order == "help")
+				to_chat(src, SPAN_NOTICE("<br>Orders give a buff to nearby soldiers for a short period of time, followed by a cooldown, as follows:<br><B>Move</B> - Increased mobility and chance to dodge projectiles.<br><B>Hold</B> - Increased resistance to pain and combat wounds.<br><B>Focus</B> - Increased gun accuracy and effective range.<br>"))
+				return
+			if(order == "cancel")
+				return
+
+	if(order)
+		if(current_aura == order)
+			visible_message(SPAN_BOLDNOTICE("[src] whitdraws their order to [order]!"), SPAN_BOLDNOTICE("You withdraw your order to [order]!"))
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/powers/issue_order.dm:33]: withdraws

File: code/modules/mob/living/carbon/human/life/handle_orders.dm [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+	var/aura_center = null
+	if(current_aura)
+		aura_center = src
+
+	if((src.job == JOB_SQUAD_LEADER || HAS_TRAIT(src, TRAIT_SOURCE_SQUAD_LEADER)) && src.assigned_squad && src.assigned_squad.num_tl) //If the guy giving orders is leading a squad with FTLs we need them to act as beacons
+		for(var/mob/living/carbon/human/marine in src.assigned_squad.ftl_list)
+			marine.handle_ftl_orders(marine)
+
+	if(aura_strength > 0)
+		for(var/mob/living/carbon/human/H as anything in GLOB.alive_human_list)
```
Comment (Zonespace27) [inline code/modules/mob/living/carbon/human/life/handle_orders.dm:12]: this is really expensive to call on every human every life() tick

it's more acceptable for xenoes bc they're rarely dealing with 30+ xenoes, but humans regularly breach 100, along with a not-exactly-free get_dist check

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by Zonespace27** [comment]
Ping me when you're ready to work on this again

**Pull request closed without merging**